recipes/newrelic/apm/go/windows.yml

```diff
@@ -13,11 +13,9 @@ installTargets:
 keywords:
   - go
 
-processMatch:
-  - go\.exe
+processMatch: []
```

Comment:
By doing this, isn't this recipe going to run all the times?

Reply:
Oh nope, `processMatch` is required in our schema. So this just makes it not return in recommendations. This is what we do in the [infra recipes](https://github.com/newrelic/open-install-library/blob/main/recipes/newrelic/infrastructure/amazonlinux2.yml#L21) as well.

Reply:
And [logs](https://github.com/newrelic/open-install-library/blob/main/recipes/newrelic/infrastructure/logs/logs.yml#L16) as well

Reply:
Right, but the problem I'm seeing, without any entry in process match, this recipe is going to run all the time, although there is nothing for it to do, and if someone happen to actually add something to do, it will run on all linux/windows hosts, which would be bad.
I think we should just remove this recipe until we see the need for it.

Reply:
> without any entry in process match, this recipe is going to run all the time

I don't believe that's how it works currently. For guided install, recipes only run if they're returned from the recommendations endpoint. With `processMatch: []` defined, the recipe doesn't have a match, so it won't return from the recommendations endpoint. 

For example: [GraphiQL example with mysqld](https://api.newrelic.com/graphiql?#query=%7B%0A%20%20docs%20%7B%0A%20%20%20%20openInstallation%20%7B%0A%20%20%20%20%20%20recommendations%28criteria%3A%20%7BinstallTarget%3A%20%7Bos%3A%20LINUX%2C%20type%3A%20HOST%7D%2C%20processDetails%3A%20%7Bname%3A%20%22mysqld%22%7D%7D%29%20%7B%0A%20%20%20%20%20%20%20%20count%0A%20%20%20%20%20%20%20%20results%20%7B%0A%20%20%20%20%20%20%20%20%20%20name%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A) doesn't return the infra or logs recipes, and they have an empty processMatch.

The same is true even with an empty `processDetails`: [GraphiQL with empty processDetails](https://api.newrelic.com/graphiql?#query=%7B%0A%20%20docs%20%7B%0A%20%20%20%20openInstallation%20%7B%0A%20%20%20%20%20%20recommendations%28criteria%3A%20%7BinstallTarget%3A%20%7Bos%3A%20LINUX%2C%20type%3A%20HOST%7D%2C%20processDetails%3A%20%7Bname%3A%20%22%22%7D%7D%29%20%7B%0A%20%20%20%20%20%20%20%20count%0A%20%20%20%20%20%20%20%20results%20%7B%0A%20%20%20%20%20%20%20%20%20%20name%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A)

Reply:
It's fine to remove these files, I think just in general it's more helpful to understand that an empty processMatch shouldn't return from recommendations and we should be able to use that if the case arises in the future.